### PR TITLE
React router

### PR DESCRIPTION
**Pull request by RamyAlshurafa**

### CODE REVIEW

File: client/src/Components/App.js [hunk 1]
```diff
@@ -1,16 +1,48 @@
...
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import SEProfile from './Pages/SEProfile';
+import HomePage from './Pages/HomePage';
+import ResultPage from './Pages/ResultPage';
+import Signup from './Pages/Signup';
+import Header from './CommonComponents/Header';
 import './App.css';
-import SEProfile from './Pages/SEProfile'
+
+
```
Comment (sultanassi95) [inline client/src/Components/App.js]: there are 3 white lines, u only should have one.

File: client/src/Components/App.js [hunk 2]
```diff
@@ -1,16 +1,48 @@
...
 
 class App extends Component {
+  state={
+    isLogin:true,
```
Comment (sultanassi95) [inline client/src/Components/App.js]: is the default value of `isLogin` key is supposed to be `true` ?

File: client/src/Components/App.js [hunk 3]
```diff
@@ -1,16 +1,48 @@
...
+import ResultPage from './Pages/ResultPage';
+import Signup from './Pages/Signup';
+import Header from './CommonComponents/Header';
 import './App.css';
-import SEProfile from './Pages/SEProfile'
+
+
 
 class App extends Component {
+  state={
```
Comment (sultanassi95) [inline client/src/Components/App.js]: please add one space between `state` ,  `=` and `{`
` state = { `

File: client/src/Components/App.js [hunk 4]
```diff
@@ -1,16 +1,48 @@
...
+    isLogin:true,
+    businessRole:'SE',
+    businessId:'',
+    businessName:'ramy company',
+    avatarUrl:'https://upload.wikimedia.org/wikipedia/commons/6/62/USPHS_Commissioned_Corps_insignia.png'
+    }
+    loginLogout=()=>{
```
Comment (sultanassi95) [inline client/src/Components/App.js]: can u explain what this function is meant for?
also add spaces, `loginLogout = () => { `

File: client/src/Components/App.js [hunk 5]
```diff
@@ -1,16 +1,48 @@
...
+              <Route path='/signup' component={Signup} exact />
+              </div>
+            </Switch>
+        </div>
+      </BrowserRouter>
     );
   }
 }
 
+
```
Comment (sultanassi95) [inline client/src/Components/App.js]: remove this white line, you have 2, u only need one.

File: client/src/Components/App.js [hunk 6]
```diff
@@ -1,16 +1,48 @@
 import React, { Component } from 'react';
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import SEProfile from './Pages/SEProfile';
+import HomePage from './Pages/HomePage';
+import ResultPage from './Pages/ResultPage';
+import Signup from './Pages/Signup';
+import Header from './CommonComponents/Header';
 import './App.css';
```
Comment (sultanassi95) [inline client/src/Components/App.js]: separate `import` statements from each other.
such as, libraries `import` statements are separated from components `import` statement with one white line.
and components `import` statements are separated from css `import` statement with one white line.

File: client/src/Components/CommonComponents/Header/index.js [hunk 7]
```diff
@@ -1,60 +1,100 @@
...
-      document.querySelector('#main').classList.add('scrolled');
-    } else {
-      document.querySelector('#navbar').classList.remove('scrolled');
-      document.querySelector('#main').classList.remove('scrolled');
-    }
-  }
-  componentDidMount() {
-    window.addEventListener('scroll', () => this.handleScroll());
+  state={
+    redirect:false
```
Comment (sultanassi95) [inline client/src/Components/CommonComponents/Header/index.js]: add one space after `redirect:`

File: client/src/Components/CommonComponents/Header/index.js [hunk 8]
```diff
@@ -1,60 +1,100 @@
...
   }
+
+  fixNav=()=>{
```
Comment (sultanassi95) [inline client/src/Components/CommonComponents/Header/index.js]: add spaces where they belong.

File: client/src/Components/CommonComponents/Header/index.js [hunk 9]
```diff
@@ -1,60 +1,100 @@
...
+       this.props.loginLogout()
+       window.removeEventListener('scroll',this.handleScroll)
+     });
+   }
+
   render() {
+    const { businessName, businessRole, isLogin ,avatarUrl } = this.props;
+
     return (
+      this.state.redirect?
```
Comment (sultanassi95) [inline client/src/Components/CommonComponents/Header/index.js]: no logic should live inside the `render` method.

File: client/src/Components/Pages/HomePage/Testimoinals/index.js [hunk 10]
```diff
@@ -6,14 +6,15 @@ import test from './test.svg';
...
 class Testimoinals extends React.Component {
-  const settings = {
-    dots: true,
-    infinite: true,
-    speed: 500,
-    slidesToShow: 1,
-    slidesToScroll: 1
 
   render() {
+    const settings = {
```
Comment (sultanassi95) [inline client/src/Components/Pages/HomePage/Testimoinals/index.js]: should be outside the `render` method.

File: client/src/Components/Pages/HomePage/index.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+class HomePage extends Component {
+  render() {
+    return (
+      <div >
+        <HomeHeader />
+        <HeroSection />
+        <HowItWork />
+        <Testimoinals />
+
```
Comment (sultanassi95) [inline client/src/Components/Pages/HomePage/index.js]: remove this line.

File: client/src/Components/App.js (hunk 4 above)
Comment (RamyAlshurafa) [reply client/src/Components/App.js]: it toggles the login state

**Pull request merged**